joystream-node.Dockerfile

```diff
@@ -1,25 +1,50 @@
 FROM rust:1.61.0-buster AS rust
+WORKDIR /joystream
 RUN rustup self update
 RUN rustup install nightly-2022-05-11 --force
 RUN rustup default nightly-2022-05-11
 RUN rustup target add wasm32-unknown-unknown --toolchain nightly-2022-05-11
 RUN rustup component add --toolchain nightly-2022-05-11 clippy
+RUN cargo install cargo-chef
 RUN apt-get update && \
   apt-get install -y curl git gcc xz-utils sudo pkg-config unzip clang llvm libc6-dev cmake
 
+FROM rust AS planner
+LABEL description="Cargo chef prepare"
+WORKDIR /joystream
+COPY Cargo.toml .
+COPY Cargo.lock .
+COPY bin ./bin
+COPY runtime ./runtime
+COPY runtime-modules ./runtime-modules
+RUN cargo chef prepare --recipe-path /joystream/recipe.json
+
+FROM rust AS cacher
+LABEL description="Cargo chef cook dependencies"
+WORKDIR /joystream
+COPY --from=planner /joystream/recipe.json /joystream/recipe.json
+ENV WASM_BUILD_TOOLCHAIN=nightly-2022-05-11
+# Build dependencies - this is the caching Docker layer!
+RUN cargo chef cook --release --recipe-path /joystream/recipe.json
```

Comment:
Is the `recipe.json` set up manually, if so should we vc it?

Reply:
the recipie.json file is produced in the previous docker layer "planner" with the command `cargo chef prepare ..` It is copied into this layer with the `COPY` command on line 25.